### PR TITLE
Cleanup data module and add docstrings

### PR DESCRIPTION
**Pull request by conorsim**
- Adding missing docstrings in `data` module
- Remove `fiftyone_plugins`
- Remove some deprecated functions

### CODE REVIEW

File: src/luxonis_ml/data/augmentations.py
Comment (conorsim) [inline src/luxonis_ml/data/augmentations.py:1]: @klemen1999 Can you add epytext docstrings for `DeterministicMosaic4` and `MixUp` and also add typing to `MixUp`?

Comment (klemen1999) [reply src/luxonis_ml/data/augmentations.py:1]: Done

File: src/luxonis_ml/data/utils/data_utils.py [hunk 1]
```diff
@@ -134,6 +129,17 @@ def _check_valid_array(path: str) -> bool:
 def transform_segmentation_value(
     value: Tuple[int, int, Union[bytes, List[int]]]
 ) -> str:
+    """Transforms a segmentation in RLE format to the format stored by
```
Comment (tersekmatija) [inline src/luxonis_ml/data/utils/data_utils.py:132]: Should we specify what our format is?

**Review by tersekmatija — APPROVED** [review]

**Review by kozlov721 — APPROVED** [review]
LGTM

**Pull request merged**